### PR TITLE
feat(nuxt): namespace global variables for multi-app support

### PR DESCRIPTION
**Pull request by nicolaspayot**
### 🔗 Linked issue

2nd task of #21635. Also linked to #25336 although it doesn't completely remove `window.__NUXT__`.

### 📚 Description

This PR updates `window.__NUXT__` global variable to handle multi-app payloads and config by storing the data under their `appId`.

Before:
```ts
window.__NUXT__ = { config: {...}, data: {...}, ...}
```
After:
```ts
window.__NUXT__ = {
  'nuxt-app-1': { config: {...}, data: {...}, ...} }
}
```
It also replaces scripts global IDs `__NUXT_DATA__` and `__NUXT_LOGS__` with data attributes:

Before:
```html
<script type="application/json" id="__NUXT_DATA__">...</script>
<script type="application/json" id="__NUXT_LOGS__">...</script>
```
After:
```html
<script type="application/json" data-nuxt-data="nuxt-app-1">...</script>
<script type="application/json" data-nuxt-logs="nuxt-app-1">...</script>
```

This will eventually allows the use of multiple Nuxt applications on a same page.

### 📝 To do:
- [x] Protect these changes under a `future` / `experimental` flag
- [x] Handle `window.__NUXT__` update in `nuxt/test-utils`. See https://github.com/nuxt/test-utils/pull/858

### CODE REVIEW

File: packages/nuxt/src/app/plugins/revive-payload.client.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export default defineNuxtPlugin({
     }
     Object.assign(nuxtApp.payload, await nuxtApp.runWithContext(getNuxtClientPayload))
     // For backwards compatibility - TODO: remove later
```
Comment (manniL) [inline packages/nuxt/src/app/plugins/revive-payload.client.ts:52]: I'm still wondering *if* we have to set the payload here / why it was marked for compat only?

File: packages/nuxt/src/core/templates.ts [hunk 2]
```diff
@@ -409,6 +409,7 @@ export const nuxtConfigTemplate: NuxtTemplate = {
       `export const vueAppRootContainer = ${ctx.nuxt.options.app.rootId ? `'#${ctx.nuxt.options.app.rootId}'` : `'body > ${ctx.nuxt.options.app.rootTag}'`}`,
       `export const viewTransition = ${ctx.nuxt.options.experimental.viewTransition}`,
       `export const appId = ${JSON.stringify(ctx.nuxt.options.appId)}`,
+      `export const multiApp = ${!!ctx.nuxt.options.future.multiApp}`,
```
[the remark this answers is not in the document]
Comment (nicolaspayot) [reply packages/nuxt/src/core/templates.ts:412]: Sure. `runningMultiApp` sounds good to me. What do you think?

File: packages/nuxt/src/core/templates.ts [hunk 3]
```diff
@@ -287,11 +287,20 @@ declare module 'nitropack/types' {
...
+      (!nuxt.options.future.multiApp
+        ? 'window?.__NUXT__?.config || window?.useNuxtApp?.().payload?.config'
+        : `window?.__NUXT__?.[${appId}]?.config || window?.useNuxtApp?.(${appId}).payload?.config`)
+        || {},
+    ].join('\n')
+  },
 }
 
+// export const useRuntimeConfig = () => ${!nuxt.options.future.multiApp ? `window?.__NUXT__?.config` : `window.__NUXT__?.[${JSON.stringify(nuxt.options.appId)}]?.config`} || {}
+
```
Comment (danielroe) [inline packages/nuxt/src/core/templates.ts:303]:
```suggestion
```

Comment (nicolaspayot) [reply packages/nuxt/src/core/templates.ts:303]: Woops, some leftover. Thank you 🙌

**Review by danielroe — APPROVED** [review]

**Comment by typed-sigterm** [comment]
I have an idea - just an idea - perhaps we don't need to differentiate between the single app and multiple apps scenarios, as a single app is just a special case of multiple apps.

**Comment by danielroe** [comment]
Agreed, I would be happy to merge that just into v4.

**Pull request merged**